nft_ingester/src/tasks/mod.rs

```diff
@@ -191,6 +191,26 @@ impl TaskManager {
             .map_err(|e| e.into())
     }
 
+    pub async fn get_task_queue_depth(conn: &DatabaseConnection) -> Result<u64, IngesterError> {
+        tasks::Entity::find()
+            .filter(
+                Condition::all()
+                    .add(tasks::Column::Status.ne(TaskStatus::Success))
```

Comment:
```suggestion
                    .add(tasks::Column::Status.eq(TaskStatus::Pending))
```
not equal to success will also count `failed` tasks which doesn't account for the queue depth

Reply:
Good catch!

Reply:
What about `Running` tasks? Don't they also contribute to queue depth?